### DISCUSSION TITLE
No releases anymore?

### DISCUSSION THREAD

**Discussion by aterjung** [OpenMage/magento-lts#2043] [Questions & Answers] [answered]
Hi community,

if im right, we don`t have a new release for more than a half year? What is the reason for this. Is it lack of time or a conscious decision?

**Comment by Flyingmana** [comment]
Hi, I can only speak for myself, but there its because of a lack of time, or rather a lack of usable time.
I had to realize last year, that my health/mental health reached a state, where I had problems to constantly concentrate on a single thing for more then an hour. Which causes issues, because the current release process when done carefully, takes several hours of concentrated work.

Therefore, a while back, I made a plan to reduce the load the release process causes, and make it a bit more flexible.
you can find it here: https://github.com/OpenMage/magento-lts/issues/2009
The biggest change is the introduction of release branches, so newly merged PRs can not conflict with the Release process, in case it takes a bit longer, or we need a bit extra time for merging security related changes.
As there were no complaints yet about the proposal, its probably safe to start the next release like described there.

Still, getting support with scripting of some of the steps would really help, as Iam not sure when I will be able to take these part on, and it would help a lot.

Also, Iam good, Iam not in a state where one needs to worry. Iam getting better. But its taking time, and the state of the world is not helping.

tl;dr: its available time. You can help here with contributing to the linked Ticket, and/or taking over some of the steps there.

**Comment by addison74** [comment]
This project is already starting to lose interest. The latest version of OpenMage was released last August. It's been 8 months since then, never before. It would be a shame to pull the shutters.

**Reply by fballiano** [reply]
I guess I'd move my customer's projects from "release" to the last commit on the 20.x branch.

It should be something "better not to be done" but on this beloved project everything moves so slowly that I don't see a problem in that.

Last commit on the 20.x branch was 15th Feb, more than 2 months ago. something to think about.

**Reply by addison74** [reply]
It would have been ideal if at least all the PR's and reported issues had been analyzed and solved. You even proposed a viable solution in the case of PR to accelerate their integration but nothing materialized. There will probably be a few left who will fork over to the latest version and then add code from time to time to fix the problems.

**Reply by fballiano** [reply]
yes it's a bit sad that that topic (trying to speed things up) ended up into nothing...

**Reply by tmotyl** [reply]
The current blocker to solve to be able to merge any PR or do a release is to fix compatibility of composer installer with newer composer version. 
The fix is already merged in composer installer https://github.com/Cotya/magento-composer-installer/commits/main but is not yet released.

So if you want to help -> please test the master branch of composer-installer in your projects and check if its still working.

**Reply by tmotyl** [reply]
one more patch in this area waiting for review https://github.com/Cotya/magento-composer-installer/pull/202

**Reply by fballiano** [reply]
@tmotyl how can I test it? I've never used M1 with composer

@ADDISON74 I've been approved as a maintainer this morning (thank you @Flyingmana), I'll try to put some energy into our beloved OM

**Reply by addison74** [reply]
@fballiano - I'm glad for your new assignment in this project. You really have all the knowledge you need to help us move forward. So far I've worked very well with you. Personally, I do not plan to change the platform too soon as long as OM works very well where I installed it and manage it.

**Reply by tmotyl** [reply]
> one more patch in this area waiting for review [Cotya/magento-composer-installer#202](https://github.com/Cotya/magento-composer-installer/pull/202)

merged.

**Comment by tmotyl** [comment]
Hi
As composer installer is now green again, and there were several patches merged to 1.9.4.x, can we plan a 1.9.4. and 20. releases? 
I think we should do a release before a security release, to make security release smaller, less risky. 
@Flyingmana can you run the release (or maybe we can do it together during video call -> to share the knowledge)? 
Is the patch https://github.com/OpenMage/magento-lts/issues/2009 a blocker for the release?

**Reply by fballiano** [reply]
I'd love to have the release process recorded in a video for future help.

Sorry I was absent this last week but I was/am dealing with flu :-\

**Reply by Flyingmana** [reply]
@fballiano I actually did record one of the Releases already :) thanks for reminding to share it https://youtu.be/TNgvvXJRiJo

@tmotyl not a blocker, just what would make it easier/less work. 
a "full" process with all steps is also listed in https://github.com/OpenMage/organizational-internal/blob/main/release-process.md
If its on the weekend or evenings (german time), I can make myself available to assist

**Reply by alexh-swdev** [reply]
Wow, more than one hour per deploy? I never expected it to be such a long process.
So again: Thanks for your time (and I am sorry I simply have none to be of any assistance)!

**Comment by luigifab** [comment]
I tried to update our website to test all changes for the next release.
First I fetch and merge 1.9.4.x and 20.0 of my fork with Github:
![image](https://user-images.githubusercontent.com/31816829/168268825-f50f4738-0626-4f53-a102-cd220d0c9aeb.png)

Then I `git pull` my local clone, `git checkout 20.0`, `git pull`, `git merge --no-ff 1.9.4.x`. Some conflict to resolve.

-----

Due to amount of changes, I did the following before apply the update on our website managed with composer:

* `cd web/` (our `magento-root-dir`)
* `git status`: the working copy is clean

Apply #1834 and #1846:
* `find app/ lib/Varien/ lib/Magento/ lib/Mage/ -name "*.xml" -print0 | xargs -0 geany`
* Replace "<?xml" by "<?xml", then save all, then close all, then quit (my Geany is configured to trim tailling whitespace on save)
* `find app/ lib/Varien/ lib/Magento/ lib/Mage/ -name "*.php" -print0 | xargs -0 geany`
* Replace "<?php" by "<?php", then save all, then close all, then quit
* `find  lib/Zend/  -name "*.php" | xargs geany`
* Replace "<?xml" by "<?xml", then save all, then close all, then quit
* `find  lib/Zend/  -name "*.xml" | xargs geany`
* Replace "<?php" by "<?php", then save all, then close all, then quit
* `git diff --ignore-all-space app/code/core/ lib/Varien/ lib/Magento/ lib/Mage/ lib/Zend/ app/design/adminhtml/default/default/layout/bundle.xml app/design/adminhtml/default/default/layout/catalog.xml app/design/adminhtml/default/default/layout/centinel.xml app/design/adminhtml/default/default/layout/downloadable.xml app/design/frontend/base/default/layout/cms.xml app/design/frontend/base/default/layout/weee.xml app/design/frontend/rwd/default/layout/cms.xml`
* `git add app/code/core/ lib/Varien/ lib/Magento/ lib/Mage/ lib/Zend/ app/design/adminhtml/default/default/layout/bundle.xml app/design/adminhtml/default/default/layout/catalog.xml app/design/adminhtml/default/default/layout/centinel.xml app/design/adminhtml/default/default/layout/downloadable.xml app/design/frontend/base/default/layout/cms.xml app/design/frontend/base/default/layout/weee.xml app/design/frontend/rwd/default/layout/cms.xml`
* `git checkout app/code/local/ app/code/community/`
* Check if there are other changes, then `git checkout` to revert them
* `git commit`
* `git status`: the working copy is clean

Apply #1995:
* `grep -Hrnl "Retrive " app/code/core/ | xargs geany`
* Replace (case sensitive) "Retrive " by "Retrieve ", then save all, then close all, then quit
* `grep -Hrnl "retrive " app/code/core/ | xargs geany`
* Replace (case sensitive) "retrive " by "retrieve ", then save all, then close all, then quit
* `grep -Hrnl "Retrive " lib/Zend/ | xargs geany`
* Replace (case sensitive) "Retrive " by "Retrieve ", then save all, then close all, then quit
* `grep -Hrnl "retrive " lib/Zend/ | xargs geany`
* Replace (case sensitive) "retrive " by "retrieve ", then save all, then close all, then quit
* `git diff`
* `git commit`
* `git status`: the working copy is clean

-----

Now I copy/paste `app/ js/ lib/ shell/ skin/` from branch 20.0 of my OpenMage fork to our website.
Because I applied #1835, I run:
* `find web/app/code/core/ web/shell/ web/lib/Varien/ web/lib/Magento/ web/lib/Mage/ web/lib/Zend/ web/lib/Unserialize/ -name "*.php" -exec php convert.php -w "{}" \;`

Then I applied **our** other changes (don't do the same):
* `rm web/js/lib/jquery/jquery-1.10.2.js`
* `rm web/js/lib/jquery/jquery-1.10.2.min.js`
* `rm web/js/lib/jquery/jquery-1.10.2.min.map`
* `rm web/js/tiny_mce/plugins/media/moxieplayer.swf`
* `rm web/skin/adminhtml/default/openmage/favicon.ico`
* `rm web/app/locale/*/template/email/admin_password_new.html`
* `rm web/js/index.php`
* `rm -r web/app/design/adminhtml/default/default/etc/`
* `rm -r web/app/design/adminhtml/default/default/locale/`
* `rm -r web/app/design/install/`
* `rm -r web/skin/install/`
* `rm -r web/skin/adminhtml/default/default/images/paypal web/app/code/core/Mage/Paypal web/app/code/core/Mage/PaypalUk`
* `rm -r web/app/design/adminhtml/default/default/template/paypal web/app/design/frontend/base/default/template/paypal web/app/design/frontend/rwd/default/template/paypal`
* `rm -r web/skin/adminhtml/default/default/images/paypal*`
* `rm web/app/locale/en_US/Mage_Paypal.csv web/app/locale/en_US/Mage_Paypaluk.csv`
* `rm web/skin/frontend/rwd/default/scss/module/_paypal.scss`
* `rm web/app/design/frontend/base/default/layout/paypal.xml`
* `rm web/app/design/frontend/base/default/layout/paypaluk.xml`
* `rm web/app/design/frontend/rwd/default/layout/paypal.xml`

And:
* `git co web/skin/adminhtml/default/default/custom.css web/js/tiny_mce/tiny_mce.js web/app/design/adminhtml/default/default/locale/*/translate.csv`

And (with some conflict to resolve):
* `git apply 1-custom.diff`
* `git apply 2-ie-no-more.diff`
* `git apply 3-pull.diff`
* `git apply 4-pull-1066.diff`
* `git apply 4-pull-1379.diff`
* `git apply 4-pull-1397.diff`
* `git apply 4-pull-1442.diff`
* `git apply 4-pull-1470.diff`
* `git apply 4-pull-1739.diff`
* `git apply 4-pull-1916.diff`
* `git apply 4-pull-9999.diff` (remove mage_persistent)
* `git apply 5-company.diff`

Then, the `git status` is small, now I checking changes before test them.

**Reply by fballiano** [reply]
it's not gonna be easy to merge everything for this next release, IMHO we should merge 1.9.4.x -> 20.x every time a PR is merged on the 1.9.4.x branch but as far as I know it's not even possible cause the branches are protected and for example I cannot commit something directly on a branch.

Is there an easy way to solve this?

**Reply by Flyingmana** [reply]
Lets first look at solving the current one.
As suggested in https://github.com/OpenMage/magento-lts/issues/2009 iam going to create separate branches for the Release building, so we can work on merging and preparing, without disrupting/conflicting with the current ongoing PRs.

lets see what is conflicting

```
Removing js/jscolor/demo.html
Auto-merging composer.json
CONFLICT (content): Merge conflict in composer.json
Auto-merging app/locale/en_US/Mage_Adminhtml.csv
CONFLICT (modify/delete): app/design/frontend/default/iphone/template/sales/order/history.phtml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/template/sales/order/history.phtml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/template/downloadable/customer/products/list.phtml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/template/downloadable/customer/products/list.phtml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/page.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/page.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/customer.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/customer.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/cms.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/cms.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/checkout.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/checkout.xml left in tree.
Auto-merging app/design/adminhtml/default/openmage/template/resetforgottenpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/resetforgottenpassword.phtml
Auto-merging app/design/adminhtml/default/openmage/template/page/header.phtml
Auto-merging app/design/adminhtml/default/openmage/template/login.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/login.phtml
Auto-merging app/design/adminhtml/default/openmage/template/forgotpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/forgotpassword.phtml
Auto-merging app/design/adminhtml/default/default/template/resetforgottenpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/resetforgottenpassword.phtml
Auto-merging app/design/adminhtml/default/default/template/page/menu.phtml
Auto-merging app/design/adminhtml/default/default/template/page/header.phtml
Removing app/design/adminhtml/default/default/template/notification/survey.phtml
Auto-merging app/design/adminhtml/default/default/template/login.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/login.phtml
Auto-merging app/design/adminhtml/default/default/template/forgotpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/forgotpassword.phtml
CONFLICT (modify/delete): app/design/adminhtml/default/default/template/example.phtml deleted in origin/release-19.4.x and modified in HEAD. Version HEAD of app/design/adminhtml/default/default/template/example.phtml left in tree.
Auto-merging app/design/adminhtml/default/default/layout/main.xml
Auto-merging app/code/core/Mage/Sales/Model/Quote.php
Auto-merging app/code/core/Mage/Core/functions.php
CONFLICT (content): Merge conflict in app/code/core/Mage/Core/functions.php
Auto-merging app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php
Removing app/code/core/Mage/Adminhtml/controllers/SurveyController.php
Auto-merging app/code/core/Mage/Adminhtml/Block/System/Config/Tabs.php
Removing app/code/core/Mage/Adminhtml/Block/Notification/Survey.php
Removing app/code/core/Mage/AdminNotification/Model/Survey.php
Automatic merge failed; fix conflicts and then commit the result.
```

lets reduce the output and categorize them.
First: conflicts where the file is not existing on one of the sides, because they are easy

```
CONFLICT (modify/delete): app/design/frontend/default/iphone/template/sales/order/history.phtml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/template/sales/order/history.phtml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/template/downloadable/customer/products/list.phtml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/template/downloadable/customer/products/list.phtml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/page.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/page.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/customer.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/customer.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/cms.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/cms.xml left in tree.
CONFLICT (modify/delete): app/design/frontend/default/iphone/layout/checkout.xml deleted in HEAD and modified in origin/release-19.4.x. Version origin/release-19.4.x of app/design/frontend/default/iphone/layout/checkout.xml left in tree.
CONFLICT (modify/delete): app/design/adminhtml/default/default/template/example.phtml deleted in origin/release-19.4.x and modified in HEAD. Version HEAD of app/design/adminhtml/default/default/template/example.phtml left in tree.
```

then the ones where the conflict is obvious and easy to resolve
```
CONFLICT (content): Merge conflict in composer.json
CONFLICT (content): Merge conflict in app/code/core/Mage/Core/functions.php
```

this leaves us with this conflicts, which are luckily not this hard to solve, but need a check of the related PRs to understand what happened/changed there on both sides.

```
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/resetforgottenpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/login.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/openmage/template/forgotpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/resetforgottenpassword.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/login.phtml
CONFLICT (content): Merge conflict in app/design/adminhtml/default/default/template/forgotpassword.phtml
```



For Ideas to prevent this. Merging it (even if automated) after each PR is likely just increase the number of conflicts. As there might be files which get modified multiple times, and then in worst case you have on each of this occurences a conflict (composer.json and composer.lock are files prone to this). And compared to what we have here after nearly 1 year of changes Iam not sure this is a big improvement.

**Reply by Flyingmana** [reply]
merge done in release-20.0 branch https://github.com/OpenMage/magento-lts/commit/7dae182c9745eaee649cbf815feb3ec8ec706793

**Reply by fballiano** [reply]
> For Ideas to prevent this. Merging it (even if automated) after each PR is likely just increase the number of conflicts. As there might be files which get modified multiple times, and then in worst case you have on each of this occurences a conflict (composer.json and composer.lock are files prone to this).

mmmm I don't know but I kinda don't agree, I think having more syncronized branches is always gonna be better, first of all because people could test a "dev-master" version of the v20 branch, which is now totally impossible, and second because it would be more easy (IMHO) to manage small conflicts then a release-wide merge, but mostly for me it's the first point, being able to have a updated version without waiting for a release is killer.

> And compared to what we have here after nearly 1 year of changes Iam not sure this is a big improvement.

that is true, we're not having a lot of conflicts for this long development cycle, but also because 90% of the PR are on the v19 branch, only more BC PR are going to be implemented I think more conflicts will come.

Question, will release-20 branch be merged into the 20.0 branch? otherwise they'll keep diverging forever no?

**Reply by Flyingmana** [reply]
> Question, will release-20 branch be merged into the 20.0 branch? otherwise they'll keep diverging forever no?

yes, directly after the release it should be merged back into the 20.0 branch.

**Reply by fballiano** [reply]
@Flyingmana do you want me to do the release so we can merge other PRs after? now that you've done the branch I think it's just a matter of tagging it from the github interface right?

**Reply by Flyingmana** [reply]
would be great if you can do it, but there are still a few steps left before tagging (see the previous linked doc)
like changing the version inlined in the code (the version bump commit) and if possible also updating the changelog, although this can also be done later

**Reply by fballiano** [reply]
right, I've created https://github.com/OpenMage/magento-lts/pull/2088 for the version bump (although I think we could reset the "patch" number and increase the "minor") cause I can't directly commit on any branch. Problem is, once merged, you'll have to merge it again in the release-20 branch.

To create the changelog I would use the github autogenerated one once the release is tagged, if that's ok with you :-)

**Reply by Flyingmana** [reply]
you can not commit directly to the release branch? that should be different :/
what message do you get?

**Reply by fballiano** [reply]
You're right, I can commit on the release branches.
The new version number is now on branch 1.9.4.x, release-19.x and release-20.0

**Reply by fballiano** [reply]
And I've drafted the new releases in https://github.com/OpenMage/magento-lts/releases (dunno if you guys can see those new ones) with auto-generated changelog (which, to me, it's awesome).
Waiting for feeback before publishing

**Reply by luigifab** [reply]
There are at least one PR reverted if I remember correctly. Need to remove them from this list?

**Reply by fballiano** [reply]
Absolutely, done it! Thanks! :-)

**Reply by Flyingmana** [reply]
if you do the version bump only in release 19, and merge it to release 20 you will have 1 merge conflict less during the next release.

**Reply by fballiano** [reply]
I cherry-picked the same commit on all the branches, shouldn't generated a conflict I think

**Reply by fballiano** [reply]
Can we publish the release? am I missing some other something we need to do?

**Reply by Flyingmana** [reply]
I think it looks fine, so yes, publishing it is ok

**Reply by fballiano** [reply]
And we have the new releases available for everyone!

Now I think only you can merge the release-20 branch into the 20.0 branch :-)

Great honor for me :-)

**Reply by addison74** [reply]
Good job folks. This one includes a long list of PR's. Let's hope we don't get more issues with this release :)

**Reply by Flyingmana** [reply]
good job.
will not come to it today, but you can also take the route over a PR to merge the releases back to the branches, then you do not depend on me having time

**Reply by fballiano** [reply]
Nice, although this way it will generate a line for the next changelog that we’ll have to remember to remove.

**Comment by Flyingmana** [comment]
is the question answered now?

I would say we now have regular releases again

**Comment by sreichel** [comment, marked as answer]
Questions was asked before https://github.com/OpenMage/magento-lts/releases/tag/v19.4.16.

-> solved?
